ant/src/balances.lua

```diff
@@ -0,0 +1,60 @@
+local utils = require(".utils")
+local json = require(".json")
+
+if not Balances then
+	Balances = {}
+end
+
+local balances = {}
+
+function balances.info(msg)
+	utils.reply(json.encode({
+		Name = Name,
+		Ticker = Ticker,
+		TotalSupply = 1,
+		Logo = Logo,
+		Denomination = tostring(Denomination),
+	}))
+end
+
+function balances.transfer(msg)
+	local from = msg.From
+	local to = msg.Tags.Recipient
+
+	local transferValidty, transferValidityError = utils.validateTransfer(msg)
```

Comment:
use `pcall` and throw an error in the `validateTransfer` function

also, i'd prefer is this function accepted just the `to, from, qty` parameters with asserts on them instead of just taking the raw `msg`

local status, result = pcall(utils.validateTransfeer, from, recipient)